### PR TITLE
Adds calendar component with dynamic availability rendering

### PR DESCRIPTION
**Pull request by ecuyle**
@timluu @leananepari 
Current functionality of the calendar is dynamic availability rendering of a selected month depending on the currently booked reservations for a given offering. The functionality of selecting dates and posting the full reservation to the backend is WIP and will be completed soon. This PR is intended to break up the Calendar feature into smaller chunks for ease of review. For context, reviewers should expect to see:
* Component with Calendar closed
screen shot 2019-02-15 at 12 30 05 am

* Component with Calendar open to current month
screen shot 2019-02-15 at 12 30 12 am

* Component with Calendar open to the next month
screen shot 2019-02-15 at 12 30 18 am

Note that the styling is being saved until after the POST functionality is working, most likely in the next sprint along with the test suite.

### CODE REVIEW

File: src/components/Calendar.jsx [hunk 1]
```diff
@@ -200,16 +208,56 @@ class Calendar extends React.Component {
           </div>
         </StyledCalendarTitle>
         <CalendarHeader values={calendar[0]} />
-        <CalendarRow values={calendar[1]} handleDayClick={this.interceptDayClick} />
-        <CalendarRow values={calendar[2]} handleDayClick={this.interceptDayClick} />
-        <CalendarRow values={calendar[3]} handleDayClick={this.interceptDayClick} />
-        <CalendarRow values={calendar[4]} handleDayClick={this.interceptDayClick} />
+        <CalendarRow
```
Comment (timluu-zz) [inline src/components/Calendar.jsx:211]: Just a suggestion, you might be able to cut down some lines of code if you use map function to return CalendarRow!

File: src/components/CalendarHeader.jsx [hunk 2]
```diff
@@ -10,13 +10,27 @@ const CalendarHeader = (props) => {
...
   return (
     <StyledCalendarRow>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[0]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[1]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[2]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[3]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[4]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[5]}</StyledDayHeaderText></StyledCalendarHeader>
-      <StyledCalendarHeader><StyledDayHeaderText>{values[6]}</StyledDayHeaderText></StyledCalendarHeader>
+      <StyledCalendarHeader>
```
Comment (timluu-zz) [inline src/components/CalendarHeader.jsx:13]: Possibly cut down some lines here using the map function as well?

**Review by timluu-zz — APPROVED** [review]
LGTM! I left a small comment here and there that might help clear up a bit of space, but honestly, the code looks good to me. I really dig the modularity and how you put your helper functions in a separate file - very readable!

**Review by anlee3 — APPROVED** [review]
LGTM. Great job separating your components and styles. Your naming convention is good as well.

File: src/components/Calendar.jsx (hunk 1 above)
Comment (ecuyle) [reply src/components/Calendar.jsx:211]: Thanks, excellent point! Will implement this in a future PR.

File: src/components/CalendarHeader.jsx (hunk 2 above)
Comment (ecuyle) [reply src/components/CalendarHeader.jsx:13]: Thanks again! Will implement this in a future PR.

**Pull request merged**